src/App.tsx

```diff
@@ -7,56 +7,153 @@ import classNames from 'classnames';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
+import { User } from './types/User';
+import { getPostsByUser, getUsers, postInfo } from './api/posts';
+import { Post } from './types/Post';
 import { Loader } from './components/Loader';
+import { Comment } from './types/Comment';
 
 export const App: React.FC = () => {
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [users, setUsers] = useState<User[]>([]);
+  const [currentUserPosts, setcurrentUserPosts] = useState<Post[]>([]);
+  const [error, setError] = useState('');
+  const [showLoader, setShowLoader] = useState(false);
+  const [sideBarIsOpen, setSideBarIsOpen] = useState(false);
+  const [currentPost, setCurrentPost] = useState<Post>();
+  const [currentPostComments, setCurrentPostComments] = useState<Comment[]>([]);
+  const [showCommentsLoader, setShowCommentsLoader] = useState(false);
+
+  const getUsersFromServer = () => {
+    getUsers('users')
+      .then((fetchedUsers: User[]) => {
+        setUsers(fetchedUsers);
+        setError('');
+      })
+      .catch(() => {
+        setError('Cannot load users');
+      });
+  };
+
+  const getUserPosts = () => {
+    if (selectedUser) {
+      setShowLoader(true);
+
+      getPostsByUser(selectedUser?.id)
+        .then((fetchedPosts) => {
+          setcurrentUserPosts(fetchedPosts);
+          setError('');
+          setShowLoader(false);
+        })
+        .catch(() => {
+          setError('Cannot load user posts');
+          setShowLoader(false);
+        });
+    }
+  };
+
+  const getPostInfo = () => {
+    if (currentPost?.id) {
+      setShowCommentsLoader(true);
```

Comment:
Replace 'currentPost?.id' with 'currentPost.id' as you already check if currentPost is not null.